tests/unit/lms/resources/lti_launch_test.py

```diff
@@ -186,6 +186,16 @@ def test_h_groupid(self, lti_launch):
             == "group:d55a3c86dd79d390ec8dc6a8096d0943044ea268@TEST_AUTHORITY"
         )
 
+    def test_h_section_groupid(self, lti_launch):
+        groupid = lti_launch.h_section_groupid(
+            "test_tool_consumer_instance_guid", "test_context_id", "test_section_id"
+        )
+
+        assert (
+            groupid
+            == "group:section-720d7708a7960df6bbfcaae1f3d4ade736544dbf@TEST_AUTHORITY"
+        )
```

Comment:
I think it might be worth having a parameterised test here to prove we are sensitive to each of the inputs.

```
@pytest.mark.paramterised('args,groupid', (
    (
        ['tool_consumer_instance_guid', 'context_id', 'section_id'],
        ""group:section-..."
    ),
    (
        ['DIFFERENT', 'context_id', 'section_id'],
        ""group:section-..."
    ),
   (
        ['tool_consumer_instance_guid', 'DIFFERENT', 'section_id'],
        ""group:section-..."
    ),
    (
        ['tool_consumer_instance_guid', 'context_id', 'DIFFERENT'],
        ""group:section-..."
    ),
))
```

Reply:
I've changed the method to accept a section dict rather than a section ID: I think it makes sense for this method to be responsible for extracting the section ID (or any other parts of the raw section dict that it needs from Canvas) itself, rather than requiring the calling code to.

Also changed this test to be a parametrized one that fully tests every line of the method: you can't break a single line of the method under test without a test failing now